packages/@aws-cdk/aws-stepfunctions-tasks/lib/dynamodb/shared-types.ts

```diff
@@ -0,0 +1,191 @@
+
+  /**
+   * The response includes only the aggregate ConsumedCapacity for the operation.
+   */
+  TOTAL = 'TOTAL',
+
+  /**
+   * No ConsumedCapacity details are included in the response.
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * Determines whether item collection metrics are returned.
+ */
+export enum DynamoItemCollectionMetrics {
+  /**
+   * If set to SIZE, the response includes statistics about item collections,
+   * if any, that were modified during the operation.
+   */
+  SIZE = 'SIZE',
+
+  /**
+   * If set to NONE, no statistics are returned.
+   */
+  NONE = 'NONE',
+}
+
+/**
+ * Use ReturnValues if you want to get the item attributes as they appear before or after they are changed
+ */
+export enum DynamoReturnValues {
+  /**
+   * Nothing is returned
+   */
+  NONE = 'NONE',
+
+  /**
+   * Returns all of the attributes of the item
+   */
+  ALL_OLD = 'ALL_OLD',
+
+  /**
+   * Returns only the updated attributes
+   */
+  UPDATED_OLD = 'UPDATED_OLD',
+
+  /**
+   * Returns all of the attributes of the item
+   */
+  ALL_NEW = 'ALL_NEW',
+
+  /**
+   * Returns only the updated attributes
+   */
+  UPDATED_NEW = 'UPDATED_NEW',
+}
+
+/**
+ * Property for any key
+ */
+export interface DynamoAttribute {
+  /**
+   * The name of the attribute
+   */
+  readonly name: string;
+
+  /**
+   * The value of the attribute
+   */
+  readonly value: DynamoAttributeValue;
+}
+
+/**
+ * Class to generate projection expression
+ */
+export class DynamoProjectionExpression {
+  private expression: string[] = [];
+
+  /**
+   * Adds the passed attribute to the chain
+   *
+   * @param attr Attribute name
+   */
+  public withAttribute(attr: string): DynamoProjectionExpression {
+    if (this.expression.length) {
+      this.expression.push(`.${attr}`);
+    } else {
+      this.expression.push(attr);
+    }
+    return this;
+  }
+
+  /**
+   * Adds the array literal access for passed index
+   *
+   * @param index array index
+   */
+  public atIndex(index: number): DynamoProjectionExpression {
+    if (!this.expression.length) {
+      throw new Error('Expression must start with an attribute');
+    }
+
+    this.expression.push(`[${index}]`);
+    return this;
+  }
+
+  /**
+   * converts and return the string expression
+   */
+  public toString(): string {
+    return this.expression.join('');
+  }
+}
+
+/**
+ * Map of string to AttributeValue
+ */
+export type DynamoAttributeValueMap = {
+  [key: string]: DynamoAttributeValue;
+};
+
+/**
+ * Class to generate AttributeValue
+ */
+export interface DynamoAttributeValue {
+  /**
+   * Sets an attribute of type String. For example:  "S": "Hello"
+   */
+  readonly s?: string;
+
+  /**
+   * Sets an attribute of type Number. For example:  "N": "123.45"
+   * Numbers are sent across the network to DynamoDB as strings,
+   * to maximize compatibility across languages and libraries.
+   * However, DynamoDB treats them as number type attributes for mathematical operations.
+   */
+  readonly n?: string;
```

Comment:
@nija-at this was previously a string - I'm not sure why... is it safe to convert it to a number or was there some use case i'm overlooking here?

Reply:
Probably a miss.

Reply:
after having a closer look at it's usage, I think it was made a string to support selecting task input data and assigning it which is not otherwise possible if it were simply a number

i.e. `{ 'N.$': '$.Item.TotalCount.N' }`

Reply:
DynamoDB numbers are expressed as `string` because they can possibly be larger than the largest value that can be safely represented in JSON. I'd have to be `BigInt` (but `jsii` does not support that) or `string` (your only hope).

In such cases, the `{ N: "123456789" }` value could be, using the kind of API I've hinted at earlier, be obtained as (obviously, names just illustrative):
- `ddb.AttributeValue.numberLiteral(123456789);`
- `ddb.AttributeValue.numberFromString("123456789");`
- `ddb.AttributeValue.numberFromJsonPath("$.Item.TotalCount.N");